edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/deployment/KubernetesDeploymentMapper.cs

```diff
@@ -80,16 +80,25 @@ public void UpdateDeployment(V1Deployment to, V1Deployment from)
             to.Metadata.ResourceVersion = from.Metadata.ResourceVersion;
         }
 
-        V1Deployment PrepareDeployment(IModuleIdentity identity, KubernetesModule module, IDictionary<string, string> labels)
+        V1Deployment PrepareDeployment(IModuleIdentity identity, KubernetesModule module, IDictionary<string, string> labels, EdgeDeploymentDefinition edgeDeploymentDefinition)
         {
             string name = identity.DeploymentName();
 
             var podSpec = this.GetPod(name, identity, module, labels);
 
             var selector = new V1LabelSelector(matchLabels: labels);
             var deploymentSpec = new V1DeploymentSpec(replicas: 1, selector: selector, template: podSpec);
+            var ownerReferences = new List<V1OwnerReference>
+            {
+                new V1OwnerReference(
+                    apiVersion: edgeDeploymentDefinition.ApiVersion,
+                    kind: edgeDeploymentDefinition.Kind,
+                    name: edgeDeploymentDefinition.Metadata.Name,
+                    uid: edgeDeploymentDefinition.Metadata.Uid,
+                    blockOwnerDeletion: true)
```

Comment:
Is this required?  We are by default doing background deletion, so this shouldn't apply.

Reply:
You are right. I will remove this after testing.